server/src/actors/api/inbound_message/create_lobby.rs

```diff
@@ -13,13 +14,16 @@ pub struct CreateLobby {
 impl Handler<CreateLobby> for WebSocket {
     type Result = ();
 
-    fn handle(&mut self, msg: CreateLobby, _ctx: &mut Self::Context) -> Self::Result {
+    fn handle(&mut self, msg: CreateLobby, ctx: &mut Self::Context) -> Self::Result {
```

Comment:
All (or most of) these lobby msg handlers for web socket seems to follow the same pattern, I wonder if we could generalize this (generics, template, proc macros, ...) :thinking:

Reply:
Maybe separate issue to try to simplify them?

Reply:
Sure, separate issue to polish the codebase is fine ;)

Idea - have you tried putting the msgs in an enum? Something like
```rust
#[derive(Message)]
pub enum PlayerMsg {
    #[rtype(result = "()")]
    CreateLobby {
        name: String,
    },
    #[rtype(result = "()")]
    SendLobbyChat {
        message: String,
    }
}

impl Handler<PlayerMsg> for Player {
    type Result = ();

    fn handle(&mut self, msg: PlayerMsg, _ctx: &mut Self::Context) -> Self::Result {

        let response = match msg {
            CreateLobby { name } => server_msg::CreateLobby {
                name,
                player_id: self.session_id,
            },
            SendLobbyChat { message } => server_msg::LobbyChatBroadcast {
                player_id: self.session_id,
                message,
            }
        };

        self.server.do_send(response);
    }
}
```
...it compiles but I'm not sure if that `derive(Message)` does what you want :man_shrugging: